GX Password/MainPage.xaml

```diff
@@ -21,26 +21,34 @@
 			<TextBlock Text="GX Password" FontWeight="Bold" VerticalAlignment="Bottom" Style="{ThemeResource CaptionTextBlockStyle}" />
 		</StackPanel>
 
-		<StackPanel Grid.Row="1" Padding="10">
-			<StackPanel Padding="0,5,0,10">
-				<TextBlock x:Uid="yourPass" Text="Your New Password:" HorizontalAlignment="Left" Padding="2"/>
-				<TextBox x:Name="P" Text="" FocusVisualPrimaryBrush="{ThemeResource SystemControlBackgroundAltMediumBrush}"  />
+		<ScrollViewer Grid.Row="1" VerticalScrollBarVisibility="Auto">
+			<StackPanel Grid.Row="1" Padding="10">
+				<StackPanel Padding="0,5,0,10">
+					<TextBlock x:Uid="yourPass" Text="Your New Password:" HorizontalAlignment="Left" Padding="2"/>
+					<TextBox x:Name="P" Text="" FocusVisualPrimaryBrush="{ThemeResource SystemControlBackgroundAltMediumBrush}"  />
+				</StackPanel>
+				<StackPanel>
+					<TextBlock x:Uid="passLeng" Text="Password Length:"/>
+					<Slider x:Name="sLeng" Value="10" Maximum="50" Minimum="1"/>
+				</StackPanel>
+				<StackPanel Padding="3,0,3,0" Visibility="Visible">
+					<TextBlock x:Uid="passOpt" Text="Password Option:"/>
+					<ToggleSwitch x:Uid="sTsNumb" x:Name="tsNumb" IsOn="True" OffContent="Disabled Numbers" OnContent="Enabled Numbers" Padding="2"/>
+					<ToggleSwitch x:Uid="sTsLett" x:Name="tsLett" IsOn="True" OffContent="Disabled Letters" OnContent="Enabled Letters" Padding="2"/>
+					<ToggleSwitch x:Uid="sTsSymb" x:Name="tsSymb" OffContent="Disabled Symbols" OnContent="Enabled Symbols" Padding="2"/>
+					<ToggleSwitch x:Uid="sTsAdv" x:Name="tsAdv" OffContent="Hidden Advanced Options" OnContent="Shows Advanced Options" Padding="2" Toggled="tsAdvCheck"/>
+				</StackPanel>
+				<StackPanel x:Name="spAdv" Padding="3,0,3,6" Visibility="Collapsed">
+					<ToggleSwitch x:Uid="sTsSimi" x:Name="tsSimi" OffContent="Disabled Similar Characters" OnContent="Enabled Similar Characters" Padding="2"/>
+					<ToggleSwitch x:Uid="sTsQuotes" x:Name="tsQuotes" OffContent="Disabled Quotation Marks" OnContent="Enabled Quotation Marks" Padding="2"/>
+					<ToggleSwitch x:Uid="sTsLettUp" x:Name="tsLettUp" IsOn="True" OffContent="Disabled Uppercase Letters" OnContent="Enabled Uppercase Letters" Padding="2"/>
+					<ToggleSwitch x:Uid="sTsLettDown" x:Name="tsLettDown" IsOn="True" OffContent="Disabled Lowercase Letters" OnContent="Enabled Lowercase Letters" Padding="2"/>
+				</StackPanel>
+				<StackPanel Padding="0,5,0,10">

```

Comment:
Wouldn't be better if the buttons were outside the Scroll Viewer? If the advanced options are being showed, the user would have to scroll down to click the generate password button, go back to the text block to see the password, then scroll down again to generate a new password or copying it to the clipboard, and so on.
![screenshot 14](https://user-images.githubusercontent.com/18544507/34542439-39a703b8-f0a2-11e7-9b85-7c7f1f8a1004.png)